openslides_backend/presenter/search_users.py

```diff
@@ -100,9 +100,11 @@ def get_result(self) -> List[List[Dict[str, Any]]]:
                     if all(
                         (instance.get(field) or "").lower() == search[field]
                         for field in search_def
-                    ):
+                    ) and any(search[field] for field in search_def):
                         current_result.append(instance)
                         break
+                    elif any(search[field] for field in search_def):
+                        break
```

Comment:
Deduplicate the if-clause:
```suggestion
                    if any(search[field] for field in search_def):
                        if all(
                            (instance.get(field) or "").lower() == search[field]
                            for field in search_def
                        ):
                            current_result.append(instance)
                        break
```
Also maybe add a comment explaining the any/all statements - I found them quite confusing to understand on the first read.

Reply:
Deduplicate the any clause. Done.